addons/udes_sale_stock/models/sale_order.py

```diff
@@ -299,11 +299,11 @@ def action_confirm(self):
         """ Override to disable tracking by default """
         return super(SaleOrder, self.with_context(tracking_disable=True)).action_confirm()
 
-    def confirm_orders(self):
+    def confirm_orders(self, states=["draft"]):
         """Process sale and cancel, confirm or ignore sale orders according to
            order processing configuration.
         """
-        to_confirm = self.search([("state", "=", "draft")])
+        to_confirm = self.search([("state", "in", states)])
```

Comment:
Depending on the outcome of the other comment, is this still needed?

Reply:
The other option is to add a function that generates the domain to search orders, so it can be overriden.

Reply:
Implemented Miquel's suggestion as it looks better than my optional argument